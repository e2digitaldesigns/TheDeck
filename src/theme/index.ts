export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
