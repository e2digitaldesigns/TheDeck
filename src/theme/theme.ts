export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
