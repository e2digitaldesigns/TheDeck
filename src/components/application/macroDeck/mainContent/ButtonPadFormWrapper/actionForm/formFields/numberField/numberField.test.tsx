import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import _upperFirst from "lodash/upperFirst";

import FormFieldNumbers, { IntFormFieldNumbersProps } from "./numberField";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../../../../../../theme/theme";

const mocked = {
  name: "seconds",
  onChange: jest.fn(),
  seconds: true,
  state: {
    profileId: "profileId",
    pageId: "pageId",
    buttonPadId: "buttonPadId",
    _id: "_id",
    order: 2,
    action: "delay",
    subAction: "",
    seconds: 0,
    url: "",
    text: "",
    scene: "",
    layer: "",
    page: "",
    path: "",
    profile: ""
  }
};

let defaultProps = {
  name: mocked.name,
  onChange: mocked.onChange,
  seconds: mocked.seconds,
  state: mocked.state
};

const testSetup = (props: IntFormFieldNumbersProps) => {
  return render(
    <ThemeProvider theme={theme}>
      <FormFieldNumbers
        name={props.name}
        onChange={props.onChange}
        seconds={props.seconds}
        state={props.state}
      />
    </ThemeProvider>
  );
};

let wrapper: any = null;
beforeEach(() => {});

afterEach(() => {
  wrapper = null;
  jest.resetAllMocks();
  jest.clearAllMocks();

  defaultProps = {
    name: mocked.name,
    onChange: mocked.onChange,
    seconds: mocked.seconds,
    state: mocked.state
  };
});

describe("<FormFieldNumbers/>", () => {
  it("Should render without errors", () => {
    wrapper = testSetup(defaultProps);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should have default values", () => {
    wrapper = testSetup(defaultProps);
    const label = wrapper.getByTestId("form_field_numberField__label");
    expect(label.innerHTML).toBe(_upperFirst(mocked.name) + ":");

    const input = wrapper.getByTestId("form_field_numberField__input");
    expect(input.value).toBe(mocked.state.seconds.toString());
  });

  it("Should call onChange", () => {
    wrapper = testSetup(defaultProps);
    const input = wrapper.queryByTestId("form_field_numberField__input");
    fireEvent.change(input, { target: { value: "3500" } });
    expect(mocked.onChange).toHaveBeenCalledTimes(1);
  });

  it("Should have base options", () => {
    defaultProps.seconds = false;
    wrapper = testSetup(defaultProps);
    const options = wrapper.queryAllByTestId("form_field_numberField__options");
    expect(options[0].value).toBe("0");
  });

  it("Should multiply options by 1000", () => {
    wrapper = testSetup(defaultProps);
    const options = wrapper.queryAllByTestId("form_field_numberField__options");
    expect(options[1].value).toBe("500");
  });
});
