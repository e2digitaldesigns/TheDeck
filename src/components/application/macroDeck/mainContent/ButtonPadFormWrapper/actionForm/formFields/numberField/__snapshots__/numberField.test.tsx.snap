// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFieldNumbers/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <label
          data-testid="form_field_numberField__label"
          for="seconds"
        >
          Seconds
          :
        </label>
        <select
          class="sc-dkPtRN iRRDCl"
          data-testid="form_field_numberField__input"
          name="seconds"
        >
          <option
            data-testid="form_field_numberField__options"
            value="0"
          >
            0
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="500"
          >
            500
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="1000"
          >
            1000
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="1500"
          >
            1500
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="2000"
          >
            2000
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="2500"
          >
            2500
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="3000"
          >
            3000
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="3500"
          >
            3500
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="4000"
          >
            4000
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="4500"
          >
            4500
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="5000"
          >
            5000
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="5500"
          >
            5500
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="6000"
          >
            6000
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="6500"
          >
            6500
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="7000"
          >
            7000
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="7500"
          >
            7500
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="8000"
          >
            8000
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="8500"
          >
            8500
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="9000"
          >
            9000
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="9500"
          >
            9500
          </option>
          <option
            data-testid="form_field_numberField__options"
            value="10000"
          >
            10000
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <label
        data-testid="form_field_numberField__label"
        for="seconds"
      >
        Seconds
        :
      </label>
      <select
        class="sc-dkPtRN iRRDCl"
        data-testid="form_field_numberField__input"
        name="seconds"
      >
        <option
          data-testid="form_field_numberField__options"
          value="0"
        >
          0
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="500"
        >
          500
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="1000"
        >
          1000
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="1500"
        >
          1500
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="2000"
        >
          2000
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="2500"
        >
          2500
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="3000"
        >
          3000
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="3500"
        >
          3500
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="4000"
        >
          4000
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="4500"
        >
          4500
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="5000"
        >
          5000
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="5500"
        >
          5500
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="6000"
        >
          6000
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="6500"
        >
          6500
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="7000"
        >
          7000
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="7500"
        >
          7500
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="8000"
        >
          8000
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="8500"
        >
          8500
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="9000"
        >
          9000
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="9500"
        >
          9500
        </option>
        <option
          data-testid="form_field_numberField__options"
          value="10000"
        >
          10000
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
